Source/JavaScriptCore/runtime/JSGlobalObject.cpp

```diff
@@ -2596,6 +2599,58 @@ void JSGlobalObject::installTypedArrayConstructorSpeciesWatchpoint(JSTypedArrayV
     m_typedArrayConstructorSpeciesWatchpoint->install(vm);
 }
 
+void JSGlobalObject::installTypedArrayIteratorProtocolWatchpoint(JSObject* base, TypedArrayType typedArrayType)
+{
+    VM& vm = this->vm();
+
+    DeferTermination deferScope(vm);
+    auto catchScope = DECLARE_CATCH_SCOPE(vm);
+
+    auto absenceCondition = [&](PropertyName propertyName) {
+        PropertySlot slot(base, PropertySlot::InternalMethodType::VMInquiry, &vm);
+        bool result = base->getOwnPropertySlot(base, this, propertyName, slot);
+        RELEASE_ASSERT(!result);
+        catchScope.assertNoException();
+        RELEASE_ASSERT(slot.isUnset());
+        RELEASE_ASSERT(base->getPrototypeDirect() == m_typedArrayProto.get(this));
+        return ObjectPropertyCondition::absence(vm, this, base, propertyName.uid(), m_typedArrayProto.get(this));
+    };
+
+    ObjectPropertyCondition lengthCondition = absenceCondition(vm.propertyNames->length);
+    ObjectPropertyCondition iteratorCondition = absenceCondition(vm.propertyNames->iteratorSymbol);
+
+    if (!lengthCondition.isWatchable(PropertyCondition::EnsureWatchability) || !iteratorCondition.isWatchable(PropertyCondition::EnsureWatchability)) {
+        typedArrayIteratorProtocolWatchpointSet(typedArrayType).invalidate(vm, StringFireDetail("Was not able to set up length watchpoint."));
+        return;
+    }
+
+    RELEASE_ASSERT(!typedArrayIteratorProtocolWatchpointSet(typedArrayType).isBeingWatched());
+    typedArrayIteratorProtocolWatchpointSet(typedArrayType).touch(vm, "Set up length watchpoint.");
+
+    typedArrayPrototypeLengthAbsenceWatchpoint(typedArrayType) = makeUnique<ObjectAdaptiveStructureWatchpoint>(this, lengthCondition, typedArrayIteratorProtocolWatchpointSet(typedArrayType));
+    typedArrayPrototypeLengthAbsenceWatchpoint(typedArrayType)->install(vm);
+    typedArrayPrototypeSymbolIteratorAbsenceWatchpoint(typedArrayType) = makeUnique<ObjectAdaptiveStructureWatchpoint>(this, iteratorCondition, typedArrayIteratorProtocolWatchpointSet(typedArrayType));
+    typedArrayPrototypeSymbolIteratorAbsenceWatchpoint(typedArrayType)->install(vm);
```

Comment:
Where is the part that makes `typedArrayIteratorProtocolWatchpointSet` ensure this:
```
3. derived TypedArray prototype's [[Prototype]] is TypedArray.prototype.
```

Reply:
ObjectPropertyCondition absence also ensures `[[Prototype]]`.

Reply:
Got it.  Thanks.